src/main/java/org/springframework/data/domain/OffsetScrollPosition.java

```diff
@@ -141,7 +149,7 @@ public OffsetScrollPosition apply(int offset) {
 				throw new IndexOutOfBoundsException(offset);
 			}
 
-			return of(startOffset + offset + 1);
+			return of(startOffset + offset);
```

Comment:
How about introducing a factory method for `OffsetPositionFunction` on `OffsetScrollPosition` to hide away `offset.isInitial() ? 0 : offset.getOffset()` as in `OffsetScrollPosition.positionFunction(offset.isInitial() ? 0 : offset.getOffset())` on the calling code site?